Mathlib/Algebra/Order/BigOperators/Expect.lean

```diff
@@ -27,13 +27,19 @@ section OrderedAddCommMonoid
 variable [AddCommMonoid α] [PartialOrder α] [IsOrderedAddMonoid α] [Module ℚ≥0 α]
   {s : Finset ι} {f g : ι → α}
 
-lemma expect_eq_zero_iff_of_nonneg (hs : s.Nonempty) (hf : ∀ i ∈ s, 0 ≤ f i) :
+lemma expect_eq_zero_iff_of_nonneg (hf : ∀ i ∈ s, 0 ≤ f i) :
     𝔼 i ∈ s, f i = 0 ↔ ∀ i ∈ s, f i = 0 := by
-  simp [expect, sum_eq_zero_iff_of_nonneg hf, hs.ne_empty]
+  rw [←sum_eq_zero_iff_of_nonneg hf, expect, smul_eq_zero, or_iff_right_iff_imp,
+      inv_eq_zero, Nat.cast_eq_zero, card_eq_zero]
+  intro hs
+  rw [hs, sum_empty]
```

Comment:
This can be shorter:
```suggestion
  simp +contextual [expect, sum_eq_zero_iff_of_nonneg hf]
```
(The "contextual" feature means that if the goal is `x = y → P x`, `simp` will also try solving `P y`.)